### PR TITLE
Calculation of pH for systems in equilibrium with atmospheric carbon dioxide

### PR DESCRIPTION
**Pull request by mhvwerts**
Hello Ryan,

As discussed, here is the code for calculation of pH of a system in equilibrium with ambient air using pHcalc.

It takes the form of a subclass of `System`, called `SystemAirEquilibrated`, which adds the necessary carbon dioxide/carbonic acid equilibria. The central property is the total 'dissolved inorganic carbon' (DIC), which is adjusted by the algorithm such that the dissolved CO2 is at equilibrium with the CO2 in the air, recalculating pH in the process.

I have made initial docstrings and added further details as comments in the code. There is also a test/example script `test_air_equilibrated.py` (reminding us that atmospheric CO2 levels are rising). I will be happy to update the documentation in `README.rst` later on.

All new code is in separate files. I did make some very minor edits to the core `pHcalc.py` but these are essentially cosmetic.  As you may notice, I anticipated the re-naming of `Acid` and `Inert` but kept this compatible with the existing names.

Best wishes,
Martin

### CODE REVIEW

**Comment by rnelsonchem** [comment]
@mhvwerts Thanks so much for this very detailed PR! I have been looking it over these past couple of days, and this led to a couple of big picture thoughts...
* The new System class w/ CO2 is very cool! But unfortunately, it is specific for CO2... Wouldn't it be neat if it was more flexible for other types of gasses that participate in acid/base equilibria?
* There total dissolved carbon concentration is solved using a minimizer, and the minimizer function also uses another minimizer call. This leads to a pretty significant performance hit. Is it possible to do this same calculation with a single call to a minimizer function?

So anyway, I tried to do this using a different strategy -- a new `AcidGas` class for gasses that participate in acid/base equilibria. I created a new PR -- GH-7 -- for a skeleton version of this addition. There's a link in the PR to a GIST demonstrating the usage. I'll add some other thoughts to the PR here as well.

File: src/pHcalc/pHcalc.py
Comment (rnelsonchem) [inline src/pHcalc/pHcalc.py:1]: Hahaha! Thanks Martin. As you can see, I have some outdated habits from the old days of Python 2/3 compatibility. I will commit these edits.

File: src/pHcalc/secant.py
Comment (rnelsonchem) [inline src/pHcalc/secant.py:1]: This is really cool! It is great that you can use this to pass some previously failed scipy tests. I'm happy to add a new minimizer method here, but this very useful bit of code will most likely be lost in this esoteric package. Would a PR at Scipy would be more appropriate? Looks like you could fork the Scipy repo and almost just drop this code into this file: https://github.com/scipy/scipy/blob/v1.10.1/scipy/optimize/_zeros_py.py

File: src/test_air_equilibrated.py
Comment (rnelsonchem) [inline src/test_air_equilibrated.py:1]: Thanks for adding some tests. I should probably create some test cases for the main code base as well. I might add these tests along with some other tests in a different PR.

**Review by rnelsonchem — COMMENTED** [review]
Some file-specific comments on this PR

File: src/pHcalc/secant.py
Comment (mhvwerts) [reply src/pHcalc/secant.py:1]: I looked into this quite some time ago, but I could not figure out how exactly prepare a decent pull request. I believe that there is a protocol to be followed. The problem has been reported to Scipy (https://github.com/scipy/scipy/issues/6379#issuecomment-1482948914), but for now I use a "custom" `secant` when I need one (which is not very often).

Anyways, the secant algorithm may not be the best choice in this particular case, other options exist. Moreover, your new #7 does not need one: a Nelder-Mead minimizer (my personal favorite!) takes care of everything.

File: src/test_air_equilibrated.py
Comment (mhvwerts) [reply src/test_air_equilibrated.py:1]: I will be happy to contribute further tests/examples.

**Comment by mhvwerts** [comment]
Your new #7 indeed provides a superior method for taking into account a solution in equilibrium with a gas-phase reservoir for a species. I will use that from now on, and we may close this particular pull request.

**Pull request closed without merging**